### PR TITLE
Fix broken defaulting and validation in TokenRequest enablement

### PR DESCRIPTION
**Pull request by mikedanese**
Defaults were not being applied correctly and the entire situation was pretty confused.

/kind bug
/sig auth

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-apiserver/app/options/validation.go [hunk 1]
```diff
@@ -49,6 +51,27 @@ func validateServiceNodePort(options *ServerRunOptions) []error {
 	return errors
 }
 
+func validateTokenRequest(options *ServerRunOptions) []error {
+	errors := []error{}
+
+	enableAttempted := options.ServiceAccountSigningKeyFile == "" ||
```
Comment (liggitt) [inline cmd/kube-apiserver/app/options/validation.go]: am I reading this wrong or are all of these conditions backwards? (enable was attempted if a signing key file != "", right?)

Comment (mikedanese) [reply cmd/kube-apiserver/app/options/validation.go]: Ha, thanks. :)

File: cmd/kube-apiserver/app/options/validation.go [hunk 2]
```diff
@@ -49,6 +51,34 @@ func validateServiceNodePort(options *ServerRunOptions) []error {
 	return errors
 }
 
+func validateTokenRequest(options *ServerRunOptions) []error {
+	errors := []error{}
```
Comment (awly) [inline cmd/kube-apiserver/app/options/validation.go]: var errors []error

File: cmd/kube-apiserver/app/options/validation.go [hunk 3]
```diff
@@ -49,6 +51,34 @@ func validateServiceNodePort(options *ServerRunOptions) []error {
...
+
+	enableAttempted := options.ServiceAccountSigningKeyFile != "" ||
+		options.Authentication.ServiceAccounts.Issuer != "" ||
+		len(options.Authentication.APIAudiences) != 0
+
+	enableSucceeded := options.ServiceAccountIssuer != nil
+
+	if enableAttempted && !utilfeature.DefaultFeatureGate.Enabled(features.TokenRequest) {
+		errors = append(errors, fmt.Errorf("the TokenRequest feature is not enabled but --service-account-signing-key-file, --service-account-issuer and/or --api-audiences flags were passed"))
```
Comment (awly) [inline cmd/kube-apiserver/app/options/validation.go]: `errors.New` here and below

File: cmd/kube-apiserver/app/options/validation.go [hunk 4]
```diff
@@ -85,6 +115,9 @@ func (s *ServerRunOptions) Validate() []error {
 	if errs := s.APIEnablement.Validate(legacyscheme.Scheme, apiextensionsapiserver.Scheme, aggregatorscheme.Scheme); len(errs) > 0 {
 		errors = append(errors, errs...)
 	}
+	if errs := validateTokenRequest(s); len(errs) > 0 {
```
Comment (awly) [inline cmd/kube-apiserver/app/options/validation.go]: `errors = append(errors, validateTokenRequest(s)...)`

File: cmd/kube-apiserver/app/server.go [hunk 5]
```diff
@@ -606,6 +558,28 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
 		}
 	}
 
+	if s.ServiceAccountSigningKeyFile != "" &&
```
Comment (awly) [inline cmd/kube-apiserver/app/server.go]: don't wrap this line

File: cmd/kube-apiserver/app/server.go [hunk 6]
```diff
@@ -606,6 +558,28 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
...
+		s.Authentication.ServiceAccounts.Issuer != "" {
+		sk, err := certutil.PrivateKeyFromFile(s.ServiceAccountSigningKeyFile)
+		if err != nil {
+			return options, fmt.Errorf("failed to parse service-account-issuer-key-file: %v", err)
+		}
+		if s.Authentication.ServiceAccounts.MaxExpiration != 0 {
+			lowBound := time.Hour
+			upBound := time.Duration(1<<32) * time.Second
```
Comment (awly) [inline cmd/kube-apiserver/app/server.go]: Is this useful at all? The value is over 136 years, might as well skip capping the expiration

File: cmd/kube-apiserver/app/options/validation.go (hunk 3 above)
Comment (mikedanese) [reply cmd/kube-apiserver/app/options/validation.go]: so... much... shadowing...

**Comment by mikedanese** [comment]
@awly @cjcullen fixed.

**Pull request merged**